ptf/saiacl.py

```diff
@@ -0,0 +1,5777 @@
+                                    ip_id=105,
+                                    ip_ttl=64)
+
+            print('#### ACL \'DROP, src mac 00:22:22:22:22:22, '
+                  'in_ports[ptf_intf_1,2,3,4]\' Applied ####')
+            print('#### Sending      ', ROUTER_MAC, '| 00:22:22:22:22:22 | '
+                  '172.16.10.1 | 192.168.0.1 | @ ptf_intf 1')
+            time.sleep(5)
+            send_packet(self, self.dev_port10, pkt)
+            print('#### NOT Expecting 00:11:22:33:44:55 |', ROUTER_MAC, '| '
+                  '172.16.10.1 | 192.168.0.1 | @ ptf_intf 4')
+            verify_no_other_packets(self, timeout=1)
+            print('#### Sending      ', ROUTER_MAC, '| 00:22:22:22:22:22 | '
+                  '172.16.10.1 | 192.168.0.1 | @ ptf_intf 2')
+            send_packet(self, self.dev_port11, pkt)
+            print('#### NOT Expecting 00:11:22:33:44:55 |', ROUTER_MAC, '| '
+                  '172.16.10.1 | 192.168.0.1 | @ ptf_intf 4')
+            verify_no_other_packets(self, timeout=1)
+            print('#### Sending      ', ROUTER_MAC, '| 00:22:22:22:22:22 | '
+                  '172.16.10.1 | 192.168.0.1 | @ ptf_intf 3')
+            send_packet(self, self.dev_port12, pkt)
+            print('#### NOT Expecting 00:11:22:33:44:55 |', ROUTER_MAC, '| '
+                  '172.16.10.1 | 192.168.0.1 | @ ptf_intf 4')
+            verify_no_other_packets(self, timeout=1)
+            time.sleep(5)
+
+        finally:
+            # cleanup ACL, remove ACL group member
+            sai_thrift_remove_acl_table_group_member(
+                self.client, acl_group_egress_ip_member_id)
+
+            # unlink this ACL table from port4 object
+            sai_thrift_set_port_attribute(self.client, self.port13,
+                                          egress_acl=int(SAI_NULL_OBJECT_ID))
+
+            # cleanup ACL group, entries, tables
+            sai_thrift_remove_acl_table_group(self.client,
+                                              acl_table_group_egress)
+            sai_thrift_remove_acl_entry(
+                self.client, acl_egress_ip_entry_id)
+            sai_thrift_remove_acl_table(
+                self.client, acl_egress_ip_table_id)
+
+    def tearDown(self):
+        # cleanup mirror session
+        sai_thrift_remove_mirror_session(self.client, self.span_session)
+        # l3 part
+        sai_thrift_remove_route_entry(self.client, self.route_entry)
+        sai_thrift_remove_next_hop(self.client, self.nhop)
+        sai_thrift_remove_neighbor_entry(self.client, self.nbr_entry)
+        super(MultAclTableGroupBindTest, self).tearDown()
+
+
+@group("draft")
+class SonicACLTest(SaiHelper):
+    """
+    Simulate the ACL setup used in ACL community tests
+    Create an ACL group per port
+    Create 3 ACL tables for ipv4 and mirror types
+    Create an ACL group member per table and per group
+    Add entries in each of the tables and test ACLs work as expected
```

Comment:
This test is wrong, we should change in our bf-switch repo as well. We can support only one V4, V6 and one mirror ACL slice in Sonic. 2 or 3 V4 ACLs cannot be supported.

For every ACL test, please add stats check for the correct ACL entry. Without that, packets can be dropped by catch-all and test will pass.